packages/geolocator/tizen/src/geolocator_tizen_plugin.cc

```diff
@@ -49,200 +143,127 @@ class GeolocatorTizenPlugin : public flutter::Plugin {
+      if (result == PermissionStatus::kDeniedForever ||
+          result == PermissionStatus::kDenied) {
+        SendErrorResult("Permission denied", "Permission denied by user.");
+        return;
+      }
+      SendResult(flutter::EncodableValue(static_cast<int>(result)));
+    } catch (const PermissionManagerError &error) {
+      SendErrorResult("Operation failed", error.GetErrorString());
       return;
     }
-    result->Success(location.ToEncodableValue());
   }
 
-  void OnGetCurrentPosition(
-      std::unique_ptr<flutter::MethodResult<flutter::EncodableValue>> result) {
-    auto result_ptr = result.release();
-    TizenResult tizen_result = location_manager_->RequestCurrentLocationOnce(
-        [result_ptr](Location location) {
-          result_ptr->Success(location.ToEncodableValue());
-          delete result_ptr;
-        },
-        [result_ptr](TizenResult error) {
-          result_ptr->Error(
-              "An error occurred while requesting current location.",
-              error.message());
-          delete result_ptr;
-        });
-
-    if (!tizen_result) {
-      result_ptr->Error("Failed to call RequestCurrentLocationOnce.",
-                        tizen_result.message());
-      delete result_ptr;
+  void OnGetLastKnownPosition() {
+    try {
+      Position position = location_manager_->GetLastKnownPosition();
+      SendResult(position.ToEncodableValue());
+    } catch (const LocationManagerError &error) {
+      SendErrorResult("Operation failed", error.GetErrorString());
     }
   }
 
-  void OnListenGeolocatorServiceUpdates(
-      std::unique_ptr<flutter::EventSink<flutter::EncodableValue>>
-          &&event_sink) {
-    geolocator_service_updates_event_sink_ = std::move(event_sink);
-    TizenResult tizen_result = location_manager_->SetOnServiceStateChanged(
-        [this](ServiceState service_state) {
-          flutter::EncodableValue value(static_cast<int>(service_state));
-          geolocator_service_updates_event_sink_->Success(value);
-        });
-    if (!tizen_result) {
-      LOG_ERROR("Failed to set OnServiceStateChanged, %s.",
-                tizen_result.message().c_str());
-      geolocator_service_updates_event_sink_ = nullptr;
+  void OnGetCurrentPosition() {
+    auto result_ptr = result_.release();
+    try {
+      location_manager_->GetCurrentPosition(
+          [result_ptr](Position position) {
+            if (result_ptr) {
+              result_ptr->Success(position.ToEncodableValue());
+              delete result_ptr;
+            }
+          },
+          [result_ptr](LocationManagerError error) {
+            if (result_ptr) {
+              result_ptr->Error("Operation failed", error.GetErrorString());
+              delete result_ptr;
+            }
+          });
+    } catch (const LocationManagerError &error) {
+      if (result_ptr) {
+        result_ptr->Error("Operation failed", error.GetErrorString());
+        delete result_ptr;
+      }
```

Comment:
Please use only one error handling mechanism at a time (either error callback or exception). You can remove the try-catch block here and modify `GetCurrentPosition` accordingly.

Reply:
The error callback is the error handling of the callback function of `location_manager_request_single_location` called in `GetCurrentPosition`. As far as i know, since this callback is called asynchronously, it was not possible to throw an exception inside the callback.

There is a way to change `GetCurrentPosition`'s return type to `bool` type and not use exception. (maybe the way you want)
I thought these three methods of `LocationManager` should be implemented in the same context: `GetCurrentPosition`, `GetLastKnownPosition`, `IsLocationServiceEnabled`.
(Reduce output arguments. Use Exceptions Rather Than Return Codes(maybe clean code?))
I don't think there is any particular problem with this way. Do you still think you should fix `GetCurrentPosition` to type `bool` and remove exception throwing? If you want to change it, I'll change it. Or please let me know if there is another way

Reply:
I modified it according to your guide.